ctools/defragment_sharded_collection.py

```diff
@@ -668,7 +668,9 @@ def get_chunk_size_or_0(ch):
             center_size_kb = await get_chunk_size(c)
             # Use < 0.6 so that we do not move chunks which were split before
             if center_size_kb > target_chunk_size_kb * args.small_chunk_frac:
-                break
+                # do not break, a chunk without a size estimate might
+                # trigger this condition accidentally
+                continue
```

Comment:
I think we can do better then this.

The most generic situation is that when we cycle over the sorted chunks (line 660) we will encounter:
1.  - some chunks without estimation size
2.  - some chunks with estimated size that are considered small
3.  - some chunks with estimated size that are considered big

For sure we need to process 1 and 2 but there is no point in inspecting all the chunks in 3 if we already know they are too big.

I think we should stop when we encounter a chunk that has the `defrag_collection_est_size` field and whose size is less then `target_chunk_size_kb * args.small_chunk_frac`